### PR TITLE
Add message producer

### PR DESCRIPTION
**Pull request by meandnano**
# What
This PR implements the producer side of the [Cloudflare Queues](https://developers.cloudflare.com/queues/configuration/javascript-apis/) feature.

# Motivation
#53

### CODE REVIEW

**Comment by syumai** [comment]
I'll check this later, thanks!

**Comment by meandnano** [comment]
Just wanted to mention it: since queues require paid plan, it might be uncomfortable to test this remotely. But I now use this implementation of queues api in my worked on a paid plan it works as expected. I'll create consumer api PR once this one is merged.

File: cloudflare/queues/producer.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	}
+	return &Producer{queue: inst}, nil
+}
+
+// Send sends a single message to a queue. This function allows setting send options for the message.
+// If no options are provided, the default options are used (QueueContentTypeJSON and no delay).
+//
+//   - https://developers.cloudflare.com/queues/configuration/javascript-apis/#producer
+//   - https://developers.cloudflare.com/queues/configuration/javascript-apis/#queuesendoptions
+func (p *Producer) Send(content any, opts ...SendOption) error {
```
Comment (syumai) [inline cloudflare/queues/producer.go:65]: In Go, `Send` method with `any` content type doesn't seem useful enough.
I think `ContentType` send option should be removed and separated `Send` methods should be declared.

e.g.
```go
func (p *Producer) SendText(content string, opts ...SendOption) error {}
func (p *Producer) SendBytes(content []byte, opts ...SendOption) error {}
func (p *Producer) SendJSON(content any, opts ...SendOption) error {}
func (p *Producer) SendV8(content js.Value, opts ...SendOption) error {}
```

I'll separate current implementation after.

File: cloudflare/queues/producer.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		return err
+	}
+
+	prom := p.queue.Call("send", jsValue, options.toJS())
+	_, err = jsutil.AwaitPromise(prom)
+	return err
+}
+
+// SendBatch sends multiple messages to a queue. This function allows setting options for	each message.
+func (p *Producer) SendBatch(messages []*BatchMessage, opts ...BatchSendOption) error {
```
Comment (syumai) [inline cloudflare/queues/producer.go:86]: I think `BatchMessage` should not just be a struct with `any` values, but a common type or interface constructed in the following form:

```go
func NewTextBatchMessage(content string, opts ...SendOption) BatchMessage
func NewBytesBatchMessage(content []byte, opts ...SendOption) BatchMessage
func NewJSONBatchMessage(content any, opts ...SendOption) BatchMessage
func NewV8BatchMessage(content js.Value, opts ...SendOption) BatchMessage
```

I'll try to change the current implementation after.

**Review by syumai — APPROVED** [review]
I'll do some changes after merging this.
Thank you for great implementation!

**Comment by syumai** [comment]
@meandnano This feature has been released as v0.27.0, thanks!
https://github.com/syumai/workers/releases/tag/v0.27.0

**Pull request merged**